### DISCUSSION TITLE
Modifying MS SQL Stored Procedure

### DISCUSSION THREAD

**Discussion by Robin-Britz** [dbeaver/dbeaver#20763] [Q&A]
Firstly, thanks @dbeaver for the epic app and all the effort!
_I'm really enjoying this a lot over SQL Server Management Studio! <3_

**Goal I want to achieve:** 
1. Modify Stored Procedure - preferably in 1 step to instantly open SQL Console with the correct DB selected (in SSMS you right click the Stored Procedure, right click > modify > good to go)

**In DBeaver I am doing the following to just modify a stored procedure, it's quite a few steps:**
1. Find the Stored Procedure I want to modify on the desired DB
3. Right Click > View Procedure (or F4)
4. On the tab that opens on the right > click "Source" to view the Stored Procedure "code"
5. On the bottom right > click "Open in SQL Console" to edit the Stored Procedure
6. It opens in dbo@master by default - I use the dropdown at the top to select correct DB (even though I did step 1)

Can I setup a shortcut somehow to even just open the stored procedure in SQL Console instantly?

**Comment by E1izabeth** [comment]
It sounds like a good idea. Could you create an issue with this use case?

Currently I can propose you to use Open Database Object dialog. Maybe you will find the procedure faster, if you know the name
![image](https://github.com/dbeaver/dbeaver/assets/28875055/059a12f8-146a-4935-8447-0e24c175593b)

**Reply by barryseymour** [reply]
I have this issue as well. Your suggestion works, and is fast, but it doesn't open the source code in a standard query window. There's no option to include a USE statement, or to use CREATE OR ALTER procedure.

You could take a cue from SSMS in this regard.

**Reply by AP-Kitsu** [reply]
> It sounds like a good idea. Could you create an issue with this use case?
> 
> Currently I can propose you to use Open Database Object dialog. Maybe you will find the procedure faster, if you know the name ![image](https://private-user-images.githubusercontent.com/28875055/258113768-059a12f8-146a-4935-8447-0e24c175593b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3ODcwODUsIm5iZiI6MTcxODc4Njc4NSwicGF0aCI6Ii8yODg3NTA1NS8yNTgxMTM3NjgtMDU5YTEyZjgtMTQ2YS00OTM1LTg0NDctMGUyNGMxNzU1OTNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE5VDA4NDYyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFhNTllNDczYmU1ZGM0ODU0ZjA2YWVjYzdkZTA0ZGNmZWNjNmNiZWVmYzA0ZjgzOTU2NWJiMTlkNmM1ZWQxNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6PVnje36l8gY_ThFzynKpTB75Qp3__qPV4hig5zZ9Ws)

I think something like SSMS creating a query with ALTER already written is just much faster

**Comment by chad-k12lift** [comment]
Came here for this exact same reason. Would love a more streamlined way to handle altering stored procedures in SQL Server.

**Comment by gciapparelli** [comment]
I found this thread and lost all hope... after two years, no news :-(
DBeaver is amazing, but it's really annoying for those of us who constantly develop on its foundation to go through the difficult path to perform a simple task like modifying a procedure.

Is this behavior unique to MS SQL Server?
